sdk/Trace/Events.php

```diff
@@ -15,7 +15,7 @@ public function addEvent(
         ?API\Attributes $attributes = null,
         int $timestamp = null
     ): API\Events {
-        $this->events[] = new Event($name, $timestamp, $attributes);
+        $this->events[] = new Event($name, $timestamp ?? time(), $attributes);
```

Comment:
Notice this one, it's probably incorrect, but the type is now correct, it's `int`, but `int|null` passed.

Reply:
Note: even though this was changed, no tests failed, meaning this wasn't tested at all. This should have been caught by Infection. /cc @prondubuisi

Reply:
According to the spec:

https://github.com/open-telemetry/opentelemetry-specification/blob/06d16d7db88c3e266636f84887de61e3363b5429/specification/trace/api.md#add-events

An Event is structurally defined by the following properties:

* Name of the event.
* A timestamp for the event. Either the time at which the event was added or a custom timestamp provided by the user.
* Zero or more Attributes further describing the event.

Maybe we shouldn't default `int $timestamp = null` on line 16 and check for the value of `$timestamp` before passing `time()` to `new Event()`

Reply:
> This API takes the name of the event, optional Attributes and an **optional** Timestamp which can be used to specify the time at which the event occurred
> If no custom timestamp is provided by the user, **the implementation automatically sets the time at which this API is called** on the event.

(emphasis mine)

Since the timestamp is supposed to be optional, it must be nullable so I think that part is correct. Event itself requires the exact timestamp which is also OK, so I guess my solution seems to be close to OK?

Reply:
Agreed that your solution is okay!